flink-connectors/flink-connector-kafka-base/src/main/java/org/apache/flink/streaming/connectors/kafka/table/KafkaOptions.java

```diff
@@ -239,9 +293,25 @@ public static KafkaSinkSemantic getSinkSemantic(ReadableConfig tableOptions){
 		}
 	}
 
-	public static StartupOptions getStartupOptions(
-			ReadableConfig tableOptions,
-			String topic) {
+	public static List<String> getSourceTopics(ReadableConfig tableOptions) {
+		return tableOptions.getOptional(TOPIC).map(value ->
+			Arrays
+				.stream(value.split(","))
+				.map(String::trim)
+				.collect(Collectors.toList()))
+			.orElse(null);
+	}
+
+	public static Pattern getSourceTopicPattern(ReadableConfig tableOptions) {
+		return tableOptions.getOptional(TOPIC_PATTERN).map(value -> Pattern.compile(value)).orElse(null);
+	}
+
+	private static boolean isSingleTopic(ReadableConfig tableOptions) {
+		// Option 'topic-pattern' is regarded as multi-topics.
+		return tableOptions.getOptional(TOPIC).isPresent() && tableOptions.get(TOPIC).split(",").length == 1;
```

Comment:
The community recommend to use List ConfigOption for list values, framework will handle the parsing. This will also change to use `;` as the separator, but this is more align with other list options. You can declare a List ConfigOption by :

```java
	public static final ConfigOption<List<String>> TOPIC = ConfigOptions
			.key("topic")
			.stringType()
			.asList()
			.noDefaultValue()
			.withDescription("...");
```

Then you can call `return tableOptions.getOptional(TOPIC).map(t -> t.size() == 1).orElse(false);` here. 


Sorry for the late reminder.